src/tools/concurrency/TraceParser.java

```diff
@@ -75,6 +136,13 @@ private void parseSymbols() {
     }
   }
 
+  private void readSourceSection(final ByteBuffer b) {
```

Comment:
Ideally, the source section information should be parsed and used, which would result in implicitly documenting what each short is for.
Otherwise, I would avoid introducing comments that become easily outdated.
The trace format should be documented where it is written.
So, perhaps there should be a JavaDoc comment here linking to the writer method.